docs/application/concepts/130-traits.mdx

```diff
@@ -0,0 +1,123 @@
+
+=== When not to use this template:
+Do not use these templates for articles that give context on how
+to use/approach the topic. Instead, it would be best if you used a guide for that.
+
+Writing a guide and a concept article are not mutually exclusive. Instead,
+you should, in most cases, write both, where the concept gives a brief overview
+of what it is, and the guide is a longer article explaining best practices on using
+it.
+
+E.g.:
+Concept: Event Bus
+Guide: Interacting with the Event Bus
+
+
+=== Writing tips:
+- Write in the present tense
+- Use neutral pronouns (they/them instead of he/she and him/her)
+- Be respectful to everyone
+- Be aware of the potential for cultural misunderstandings
+-->
+
+<!-- Relevant imports: -->
+
+import { Reference, Image } from '/components';
+
+<!-- Brief Description (1-3 lines) of the concept: -->
+
+Traits add additional functionality to your Java class
+by providing default implementations you can then "import".
+
+## Basic functionality
+
+Traits are essentially nothing more than Java interfaces.
+They provide default methods to the implementing class or interface.
+You can "import" traits via the `implements` statement in your class definition.
+
+The difference between traits and normal Java interfaces are essentially that a trait
+adds functionality without forcing the implementing class
+to overwrite/implement the defined methods like normal interfaces would do.
+
+After importing a trait you have access to the provided methods from the trait
+in your current scope.
+
+Let's look on an example for a better understanding:
+
+## A very basic logging trait
+
+Let's say, we have a class that wants to log a message to the console:
+
+```java title="MyClass.java"
+public class MyClass {
+	public void addNumbers() {
+		int sum = 1 + 2;
+		// that's tedious
+		// and where's the classname in front of the logging message?
+		System.out.printf("The sum is %d", sum); 
+	}
+}
+```
+
+And we have a trait to support the requested feature:
+
+```java title="WithLogging.java"
+public interface WithLogging {
+	default void log(String format, Object ...args) {
+		System.out.print(getClass().getName() + ": ");
+		System.out.printf(format, args);
+	}
+}
+```
+
+You can now implement the `WithLogging` and simply
+using the default implemented method `log`:
+
+```java title="MyClass.java" {1,4}
+public class MyClass implements WithLogging {
+	public void addNumbers() {
+		int sum = 1 + 2;
+		log("The sum is %d", sum);
+	}
+}
+```
+
+## Conventions
+
+The name of a trait usually contains the prefix `With`
+and an action call describing what the trait provides
+to the implementing class or interface. For example:
+
+- `WithLogging`
+- `WithEventBus`
+- `WithTiming`
+- ...
+
+## See also
+
+<!--
+Snippets
+--------
+
+<Reference to="../other-article">
+		Relative Link to other article
+</Reference>
+
+<Reference to="https://www.example.com">
+		Example Website
+</Reference>
+-->
```

Comment:
<!-- GRAPHITE_METADATA {"trueLocation":{"startLine":110,"endLine":124}} -->
<i class='graphite__hidden'>[Re: lines 110 to 124]<br/><br/></i>

Let's add a link to the API Reference of all available `TelestionVerticle` traits, here, please: https://javadoc.io/doc/de.wuespace.telestion/telestion-api/latest/de/wuespace/telestion/api/verticle/trait/package-summary.html

Reply:
Done!